### PR TITLE
Normalize observations when adding experiences

### PR DESCRIPTION
**Pull request by harperj**
This change moves normalization of vector observations into the trainer's
"add_experiences" interface.

Prior to this change, normalization occurred at inference time.  This
was somewhat confusing since usually executing a forward pass shouldn't
have side-effects which would change the training step.  Also, in a
asynchronous or distributed setting where we copy the neural network
weights from a trainer to a remote actor / inference worker we'd end up
with training issues because of the weights being different on the trainer
than the workers.

**Tested on:** CrawlerDynamic with PPO and SAC.  Saw no major discrepancies in the performance  when compared against `develop`.

### CODE REVIEW

File: ml-agents/mlagents/trainers/tf_policy.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def __init__(self, seed, brain, trainer_parameters):
         self.seed = seed
         self.brain = brain
         self.use_recurrent = trainer_parameters["use_recurrent"]
+        self.normalize = trainer_parameters.get("normalize", False)
```
Comment (chriselion) [inline ml-agents/mlagents/trainers/tf_policy.py]: Don't you want this to default to True?

Comment (ervteng) [reply ml-agents/mlagents/trainers/tf_policy.py]: Default is False as normalize modifies the observation values

File: ml-agents/mlagents/trainers/ppo/policy.py [hunk 2]
```diff
@@ -176,6 +169,7 @@ def update(self, mini_batch, num_sequences):
         :param num_sequences: Number of sequences to process.
         :return: Results of update.
         """
+
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/ppo/policy.py]: Why the newlines?

File: ml-agents/mlagents/trainers/ppo/trainer.py [hunk 3]
```diff
@@ -91,6 +91,8 @@ def process_experiences(
         :param new_info: Dictionary of all next brains and corresponding BrainInfo.
         """
         info = new_info[self.brain_name]
+        if self.is_training:
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/ppo/trainer.py]: Does this crash if vector_observations is []?

File: ml-agents/mlagents/trainers/ppo/policy.py (hunk 2 above)
Comment (harperj) [reply ml-agents/mlagents/trainers/ppo/policy.py]: Should have been removed, will fix.

File: ml-agents/mlagents/trainers/ppo/trainer.py (hunk 3 above)
Comment (harperj) [reply ml-agents/mlagents/trainers/ppo/trainer.py]: It should not.  Firstly because it should be impossible to use vector observations and not return any. Second of all because the tensor is of variable length.

File: ml-agents/mlagents/trainers/tf_policy.py (hunk 1 above)
Comment (harperj) [reply ml-agents/mlagents/trainers/tf_policy.py]: I'm actually a little torn on what the default should be longer term since most users will likely need to normalize their vector obs -- but I think that for now it makes sense to stay with the more conservative choice.
